core/src/mindustry/game/MapObjectives.java

```diff
@@ -685,6 +703,54 @@ public void draw(){
 
             WorldLabel.drawAt(fetchedText, pos.x, pos.y + radius + textHeight, Draw.z(), flags, fontSize);
         }
+
+        @Override
+        public void control(LMarkerControl type, float p1, float p2, float p3){
+            switch(type){
+                case x -> pos.x = p1 * tilesize;
+                case y -> pos.y = p1 * tilesize;
+                case pos -> pos.set(p1 * tilesize, p2  * tilesize);
+                case fontSize -> fontSize = p1;
+                case textHeight -> textHeight = p1;
+                case labelBackground -> {
+                    if(p1 != 0.0){
+                        flags |= WorldLabel.flagBackground;
+                    }else{
+                        flags &= ~WorldLabel.flagBackground;
+                    }
+                }
+                case labelOutline -> {
+                    if(p1 != 0.0){
+                        flags |= WorldLabel.flagOutline;
+                    }else{
+                        flags &= ~WorldLabel.flagOutline;
+                    }
+                }
+                case labelFlags -> {
+                    flags = (p1 != 0.0 ? WorldLabel.flagBackground : 0);
+                    if(p2 != 0.0) flags |= WorldLabel.flagOutline;
+                }
+                case radius -> radius = p1;
+                case rotation -> rotation = p1;
+                case shapeSides -> sides = (int)p1;
+                default -> super.control(type, p1, p2, p3);
+            }
+        }
+
+        @Override
+        public void setText(String text, boolean fetch){
+            this.text = text;
+            if(fetch){
+                fetchedText = fetchText(this.text);
+            }else{
+                fetchedText = this.text;
+            }
+        }
+
+        @Override
+        public void setColor(double value){
```

Comment:
Why is setColor a separate method - couldn't it be part of control, if the parameters were doubles?

Reply:
Actually, earlier (even before the first commit) it used to be double, but there were tons of float typecasts in control() method, so i changed 
control()'s args type to float and created another method for controlling color as it strictly requires double value.

It's a choice between having a separate method for that (less optimized i guess) or having these typecasts in control(). Can revert this back if needed.